napari/utils/events/_tests/test_evented_model.py

```diff
@@ -461,31 +478,43 @@ def test_evented_model_with_property_setters():
     assert t.c == [2, 3]
     assert t.a == 2
     assert t.b == 3
+    t.d = 4
+    assert t.a == 2
+    assert t.b == 2
+    t.e = 100
+    assert t.a == 10
 
 
-def test_evented_model_with_property_setters_events():
+@pytest.fixture()
+def mocked_object():
     t = T()
-    assert 'c' in t.events  # the setter has an event
     t.events.a = Mock(t.events.a)
     t.events.b = Mock(t.events.b)
     t.events.c = Mock(t.events.c)
-
-    # setting t.c emits events for all three a, b, and c
-    t.c = [10, 20]
-    t.events.a.assert_called_with(value=10)
-    t.events.b.assert_called_with(value=20)
-    t.events.c.assert_called_with(value=[10, 20])
-    assert t.a == 10
-    assert t.b == 20
-
-    t.events.a.reset_mock()
-    t.events.b.reset_mock()
-    t.events.c.reset_mock()
-
-    # setting t.a emits events for a and c, but not b
-    # this is because we declared c to be dependent on ['a', 'b']
-    t.a = 5
-    t.events.a.assert_called_with(value=5)
-    t.events.c.assert_called_with(value=[5, 20])
-    t.events.b.assert_not_called()
-    assert t.c == [5, 20]
+    t.events.d = Mock(t.events.d)
+    t.events.e = Mock(t.events.e)
+    return t
+
+
+@pytest.mark.parametrize(
+    'attribute,value,expected_event_values',
+    [
+        ('a', 5, {'a': 5, 'b': None, 'c': [5, 1], 'd': 6, 'e': 50}),
+        ('b', 5, {'a': None, 'b': 5, 'c': [1, 5], 'd': 6, 'e': None}),
+        ('c', [10, 20], {'a': 10, 'b': 20, 'c': [10, 20], 'd': 30, 'e': 100}),
+        ('d', 8, {'a': 4, 'b': 4, 'c': [4, 4], 'd': 8, 'e': 40}),
+        ('e', 100, {'a': 10, 'b': None, 'c': [10, 1], 'd': 11, 'e': 100}),
+    ],
+)
+def test_evented_model_with_property_setters_events_1(
+    mocked_object, attribute, value, expected_event_values
+):
```

Comment:
From the one side, it is more compacted, from the other side, it is less readable. 

Adding doctrine with a test explanation for the current code will be nice.